src/morphology.rs

```diff
@@ -349,6 +351,567 @@ pub fn close_mut(image: &mut GrayImage, norm: Norm, k: u8) {
     erode_mut(image, norm, k);
 }
 
+/// A struct representing a mask used in morphological operations
+///
+/// The mask can have any size between 0 by 0 to 511 by 511 pixels
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Mask {
+    /// for any optimisation/arithmetic purposes, it is garanteed that :
+    ///
+    /// - all the integer values will be strictly between -512 and 512
+    /// - all tuples will be sorted in lexicographic order
+    /// - no tuple shall appear twice
+    elements: Vec<(i16, i16)>,
+}
+
+impl Mask {
+    /// creates a mask from a grayscale image
+    ///
+    /// a pixel is part of the mask if and only if it is non-zero
+    ///
+    /// `center_x` and `center_y` define the coordinates of the center of the mask.
+    /// They may take any value, including outside of the bounds of the input image,
+    /// but the image itself must be at most 511 pixel wide and 511 pixel high.
+    ///
+    /// # Panics
+    /// if the input image is 512 or more pixels wide or 512 or more pixel high
+    ///
+    /// # Example
+    /// ```
+    /// # extern crate image;
+    /// # #[macro_use]
+    /// # extern crate imageproc;
+    /// # fn main() {
+    /// use imageproc::morphology::Mask;
+    ///
+    ///
+    /// let ring_mask_base = gray_image!(
+    ///     100,  75, 255, 222;
+    ///      84,   0,   0,   1;
+    ///      99,   0,   0,  22;
+    ///     255,   7, 255,  20
+    /// );
+    ///
+    /// let ring_mask_inside = Mask::from_image(&ring_mask_base, 1, 1);
+    ///
+    /// // two different images with identical zeroes will create the same mask
+    /// let other_ring_mask_base = gray_image!(
+    ///      10, 172,  13,   5;
+    ///      45,   0,   0, 101;
+    ///     222,   0,   0,  93;
+    ///       1,   7, 212,  35
+    /// );
+    ///
+    /// assert_eq!(ring_mask_inside, Mask::from_image(&other_ring_mask_base, 1, 1));
+    ///
+    /// // using two identical images with different centers will usually make different masks
+    /// assert_ne!(Mask::from_image(&ring_mask_base, 1, 1), Mask::from_image(&ring_mask_base, 2, 2));
+    ///
+    /// // the center may be out of the image bounds
+    /// let ring_mask_outside = Mask::from_image(&ring_mask_base, 8, 8);
+    ///
+    /// // but the input image itself must be at most 511 pixel wide and 511 pixel wide.
+    /// // the code below would panic :
+    /// // let some_mask = Mask::from_image(&GrayImage::new(600, 600), 2, 2);
+    /// # }
+    /// ```
+    pub fn from_image(image: &GrayImage, center_x: u8, center_y: u8) -> Self {
```

Comment:
Are you sure about `u8`?
What about `256`, for example?

Doc: They may take any value, including outside of the bounds of the input image.

Reply:
i meant any `u8` value, that's not clear so i'll look into rewording it

Reply:
i updated it to say the value must be between 0 and 255